ops/src/purge/errored_txs_purger.rs

```diff
@@ -0,0 +1,296 @@
+            let conn = pool.connection();
+            let tree = self.tree.expect("Tree not set");
+
+            let mut paginator = cl_audits_v2::Entity::find()
+                .filter(cl_audits_v2::Column::Tree.eq(tree))
+                .select_only()
+                .columns([cl_audits_v2::Column::Tx, cl_audits_v2::Column::LeafIdx])
+                .into_model::<TxQueryResult>()
+                .paginate(&conn, batch_size);
+
+            while let Ok(Some(records)) = paginator.fetch_and_next().await {
+                if sender.send(records).is_err() {
+                    error!("Failed to send keys");
+                }
+            }
+        })
+    }
+}
+
+struct MarkDeletion {
+    rpc: Option<Rpc>,
+    receiver: Option<UnboundedReceiver<Vec<TxQueryResult>>>,
+    sender: Option<UnboundedSender<i64>>,
+    concurrency: Option<usize>,
+}
+
+impl MarkDeletion {
+    pub const DEFAULT_CONCURRENCY: usize = 10;
+
+    fn build() -> Self {
+        Self {
+            rpc: None,
+            receiver: None,
+            sender: None,
+            concurrency: None,
+        }
+    }
+
+    fn rpc(mut self, rpc: Rpc) -> Self {
+        self.rpc = Some(rpc);
+        self
+    }
+
+    fn receiver(mut self, receiver: UnboundedReceiver<Vec<TxQueryResult>>) -> Self {
+        self.receiver = Some(receiver);
+        self
+    }
+
+    fn sender(mut self, sender: UnboundedSender<i64>) -> Self {
+        self.sender = Some(sender);
+        self
+    }
+
+    fn concurrency(mut self, concurrency: usize) -> Self {
+        self.concurrency = Some(concurrency);
+        self
+    }
+
+    fn start(self) -> JoinHandle<()> {
+        let rpc = self.rpc.expect("Rpc not set");
+        let mut receiver = self.receiver.expect("Receiver not set");
+        let sender = self.sender.expect("Sender not set");
+        let concurrency = self.concurrency.unwrap_or(Self::DEFAULT_CONCURRENCY);
+
+        tokio::spawn(async move {
+            let mut tasks = JoinSet::new();
+            while let Some(chunk) = receiver.recv().await {
+                let rpc = rpc.clone();
+                let sender = sender.clone();
+                if tasks.len() >= concurrency {
+                    tasks.join_next().await;
+                }
+
+                tasks.spawn(async move {
+                    for tx_query_result in chunk {
+                        let sig = Signature::try_from(tx_query_result.tx.as_ref()).unwrap();
+                        if let Ok(tx) = rpc.get_transaction(&sig).await {
+                            if tx.transaction.meta.clone().unwrap().err.is_some() {
+                                if let Err(e) = sender.send(tx_query_result.leaf_idx) {
+                                    error!("Failed to send marked leaves {:?}", e);
+                                }
+                            }
+                        }
+                    }
+                });
```

Comment:
You loose track of this inner span and its still doing a batch of get_transactions requests. We can do this way but the upper spawn should launch all requests and then use join task to await all the results. That will make a batch of get signature calls concurrently.

Your max concurrency becomes deletion workers x signature batch size